spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -70,6 +70,21 @@ def __init__(self, *args, **kwargs):
         self._tmp_reading = False
         self._pdb_frame_loc = (None, None)
 
+    def will_close(self, externally_managed):
+        """
+        Close communication channels with the kernel if shutdown was not
+        called. If the kernel is not externally managed, shutdown the kernel
+        as well.
```

Comment:
This docstring doesn't seem to describe correctly what's happening in the code present in this method. Please update it.